pkgs/applications/networking/xpipe/default.nix

```
@@ -0,0 +1,71 @@
+{ stdenvNoCC, lib, fetchzip, makeDesktopItem }:
+
+let
+  inherit (stdenvNoCC.hostPlatform) system;
+  throwSystem = throw "Unsupported system: ${system}";
+
+  arch = {
+    x86_64-linux = "x86_64";
+    aarch64-linux = "arm64";
+  }.${system} or throwSystem;
+
+  hash = {
+    x86_64-linux = "sha256-VmJa0DqnFSzq4wJ+HchguDK4tMeKP/XaE2MidXfootM=";
+    aarch64-linux = "sha256-LdXfhZK4n8HqN1u/rKIIZhFk7fREXbTuAzyvvhFR6Lo=";
+  }.${system} or throwSystem;
+
+  displayname = "XPipe";
+
+in stdenvNoCC.mkDerivation rec {
+  pname = "xpipe";
+  version = "1.6.0";
+
+  src = fetchzip {
+    url = "https://github.com/xpipe-io/${pname}/releases/download/${version}/xpipe-portable-linux-${arch}.tar.gz";
+    inherit hash;
+  };
+
+  desktopItem = makeDesktopItem {
+    categories = [ "Network" ];
+    comment = "The shell connection hub and remote file browser for your entire infrastructure";
+    desktopName = "${displayname}";
+    exec = "/opt/${pname}/cli/bin/xpipe open %U";
+    genericName = "Shell connection hub";
+    icon = "/opt/${pname}/logo.png";
+    name = "${displayname}";
+  };
+
+  installPhase = ''
+    runHook preInstall
+
+    pkg="${pname}"
+    mkdir -p $out/opt/$pkg
+    cp -r ./ $out/opt/$pkg
+
+    mkdir -p "$out/usr/bin"
+    ln -s "$out/opt/$pkg/cli/bin/xpipe" "$out/usr/bin/$pkg"
+
+    mkdir -p "$out/usr/share/applications"
+    cp -r "${desktopItem}/share/applications/" "$out/usr/share/"
+
+    mkdir -p "$out/etc/bash_completion.d"
+    ln -s "$out/opt/$pkg/cli/xpipe_completion" "$out/etc/bash_completion.d/$pkg"
+
+    substituteInPlace $out/usr/share/applications/${displayname}.desktop --replace "Exec=" "Exec=$out"
+    substituteInPlace $out/usr/share/applications/${displayname}.desktop --replace "Icon=" "Icon=$out"
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "The shell connection hub and remote file browser for your entire infrastructure";
+    homepage = "https://github.com/xpipe-io/${pname}";
+    downloadPage = "https://github.com/xpipe-io/${pname}/releases/latest";
+    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
+    changelog = "https://github.com/xpipe-io/${pname}/releases/tag/${version}";
+    license = [ licenses.asl20 licenses.unfree ];
+    maintainers = with maintainers; [ crschnick ];
+    platforms = [ "x86_64-linux" "aarch64-linux" ];
+    mainProgram = "${pname}";
```

These string interpolations are unnecessary. Same goes for the stuff in the desktop item.

```suggestion
    mainProgram = pname;
```